pkg/ruler/ruler.go

```diff
@@ -626,11 +647,11 @@ func (r *Ruler) listRulesShuffleSharding(ctx context.Context) (map[string]rulesp
 //
 // Reason why this function is not a method on Ruler is to make sure we don't accidentally use r.ring,
 // but only ring passed as parameter.
-func filterRuleGroups(userID string, ruleGroups []*rulespb.RuleGroupDesc, ring ring.ReadRing, instanceAddr string, log log.Logger, ringCheckErrors prometheus.Counter) []*rulespb.RuleGroupDesc {
```

Comment:
From the comments it looks like it is intent to not have the method on ruler

Reply:
This appears to be just a safety measure to prevent future mistakes, and not a hard requirement.  The method still does not use `r.ring`, and the only change to its functionality is that it indirectly uses `r.ruleGroupHashFunc` via `r.instanceOwnsRuleGroup` (to allow for custom hash functions in tests).

I'll update the comment.